### DISCUSSION TITLE
4G USB MODEM HUAWEI K5161h

### DISCUSSION THREAD

**Discussion by redstamp101** [Ysurac/openmptcprouter#3583] [Q&A] [answered]
Great development btw - and really looking forward to getting it working!!

Anyone used these modems and got them to work?

- Using Rasberry Pi 4B with 2x 4G USB dongles.
- Attached into Pi USB slots directly.
- Modems (branded Vodafone) on two different networks (UK EE and UK O2) and work fine for data connectivity when plugged into laptop (and can ping internet and my VPS).
- No DHCP settings in the modem admin settings, so cannot disable, but set OMR WAN settings configured as described in the minimal set-up.
- Access OMR interface from laptop (XPS15-WIN10) over ethernet cable.

Any help or suggestions gratefully received.

![image](https://github.com/user-attachments/assets/68b970b8-e88f-493f-bd49-3d1c5077d10e)

**Comment by Ysurac** [comment, marked as answer]
There is multiple problems I think:
* File system is read only. You can try a ```e2fsck /dev/mmcblk0p2``` or reinstall it on the SD card.
* If the modem are DHCP, you should set wan1 & wan2 as DHCP client.

**Reply by redstamp101** [reply]
Great - thanks for fast reply - I did wonder why it was saying read-only. Will try suggested fixes.

**Reply by redstamp101** [reply]
Worked!  Both fixes were needed.
